### DISCUSSION TITLE
Event Handling for SignIns

### DISCUSSION THREAD

**Discussion by Megajin** [supabase/supabase#15456] [Feature Requests]
Hi there,

How do you guys handle events like a successful login? I have not seen any option to put in functions like `onSignIn` likewise for post and pre functions. I need these to do some loading indication. The `redirectTo` option seems to only work on signups?

Use case:

The user clicks on "Login", after that I want to show a loading indicator and if possible disable the button for people with slow internet connections who will click multiple times on a button. After a successful login, I'd like to use my own router in my application to redirect the user.

Packages currently in use:

```json
    "@supabase/auth-helpers-nextjs": "^0.7.2",
    "@supabase/auth-ui-react": "^0.4.2",
    "@supabase/auth-ui-shared": "^0.1.6",
    "@supabase/supabase-js": "^2.26.0",
```

**Comment by ThatLinuxGuyYouKnow** [comment]
I cant directly help as I'm not familiar with next.js.
However, it seems the resources at [Supabase Auth Helpers](https://supabase.com/docs/guides/getting-started/tutorials/with-nextjs#supabase-auth-helpers) may be useful.

**Reply by Megajin** [reply]
Hi, 

I have been working through those docs. I ended up creating my own form, as I was in need for event handling while loading. That's why I was asking how you guys handle something like "onSignInCallback". People with slow internet connections have to wait for the login, simply putting "..." behind the signin button won't do it. I need to disable that button and maybe show some other indication.

**Reply by ThatLinuxGuyYouKnow** [reply]
Hallo,

Oh, I think I understand that a lot better.
Hopefully one of the devs sees this.
